src/app/components/speakers-page/speakers-page.component.ts

```diff
@@ -4,19 +4,110 @@ import { Speakers } from 'src/app/models/interfaces';
+    selector: 'app-speakers-page',
+    templateUrl: './speakers-page.component.html',
+    styleUrls: ['./speakers-page.component.sass']
 })
 export class SpeakersPageComponent implements OnInit {
-  speakers: Speakers = <Speakers>{};
-  constructor(private db: AngularFirestore) {
-    db.doc<Speakers>('speakers/data').valueChanges().subscribe(data => {
-      this.speakers = data;
-    });
-  }
+    speakers: Speakers = <Speakers>{};
+    mobile = [];
+    web = [];
+    cloud = [];
+    ml = [];
+    others = [];
+    web_Active;
+    mobile_Active;
+    ml_Active;
+    cloud_Active;
+    others_Active;
+    webCount;
+    mobileCount;
+    mlCount;
+    cloudCount;
+    othersCount;
+    constructor(private db: AngularFirestore) {
+        db.doc<Speakers>('speakers/data').valueChanges().subscribe(data => {
+            this.speakers = {
+                enabled: data.enabled,
+                data: data.data.filter(ele => {
+                    // console.log(ele);
+                    if (ele.track == 'mobile') {
+                        console.log(ele.track);
+                        this.mobile.push(ele);
+                    }
+                    else if (ele.track == 'web') {
+                        console.log(ele.track);
+                        this.web.push(ele);
+                    }
+                    else if (ele.track == 'cloud') {
+                        console.log(ele.track);
+                        this.cloud.push(ele);
+                    }
+                    else if (ele.track == 'ml') {
+                        console.log(ele.track);
+                        this.ml.push(ele);
+                    } else {
+                        console.log(ele.track);
+                        this.others.push(ele);
+                    }
+                    return true
+                })
+            }
+        });
 
-  ngOnInit() {
-  }
+    }
+    webActive(value) {
+        this.webCount += value;
+        if (this.webCount % 2 == 0) {
+            this.web_Active = false;
+        } else {
+            this.web_Active = true;
+        }
+    }
+    mobileActive(value) {
+        this.mobileCount += value;
+        if (this.mobileCount % 2 == 0) {
+            this.mobile_Active = false;
+        } else {
+            this.mobile_Active = true;
+        }
+    }
+    mlActive(count) {
+        this.mlCount += count;
+        if (this.mlCount % 2 == 0) {
+            this.ml_Active = false;
+        } else {
+            this.ml_Active = true;
+        }
+    }
+    cloudActive(value) {
+        this.cloudCount += value;
+        if (this.cloudCount % 2 == 0) {
+            this.cloud_Active = false;
+        } else {
+            this.cloud_Active = true;
+        }
+    }
+    othersActive(value) {
+        this.othersCount += value;
+        if (this.othersCount % 2 == 0) {
+            this.others_Active = false;
+        } else {
+            this.others_Active = true;
+        }
```

Comment:
You can also write the same things as:
```js
this.someActive = this.someCount%2;
```

Also, please use consistent styles, i.e. use only camelCase here.

Reply:
instead I'll be using checkbox which will reduce some line of code and also will look much better